packages/date-picker/src/DatePicker/DatePickerInput/DatePickerInput.tsx

```diff
@@ -0,0 +1,245 @@
+import React, {
+  ChangeEvent,
+  FocusEventHandler,
+  forwardRef,
+  KeyboardEventHandler,
+  MouseEventHandler,
+} from 'react';
+
+import { createSyntheticEvent, keyMap } from '@leafygreen-ui/lib';
+
+import { DateFormField, DateInputBox } from '../../shared/components/DateInput';
+import { DateInputSegmentChangeEventHandler } from '../../shared/components/DateInput/DateInputSegment';
+import { useDatePickerContext } from '../../shared/components/DatePickerContext';
+import {
+  isElementInputSegment,
+  isExplicitSegmentValue,
+  isSameUTCDay,
+  isValidSegmentName,
+  isValidValueForSegment,
+  isZeroLike,
+} from '../../shared/utils';
+import { useSingleDateContext } from '../SingleDateContext';
+import { getRelativeSegment } from '../utils/getRelativeSegment';
+import { getSegmentToFocus } from '../utils/getSegmentToFocus';
+
+import { DatePickerInputProps } from './DatePickerInput.types';
+
+export const DatePickerInput = forwardRef<HTMLDivElement, DatePickerInputProps>(
+  (
+    {
+      onClick,
+      onKeyDown,
+      onChange: onSegmentChange,
+      ...rest
+    }: DatePickerInputProps,
+    fwdRef,
+  ) => {
+    const { formatParts, disabled, isDirty, setIsDirty } =
+      useDatePickerContext();
+    const {
+      refs: { segmentRefs, calendarButtonRef },
+      value,
+      setValue,
+      openMenu,
+      toggleMenu,
+      handleValidation,
+    } = useSingleDateContext();
+
+    /** Called when the input's Date value has changed */
+    const handleInputValueChange = (inputVal?: Date | null) => {
+      if (!isSameUTCDay(inputVal, value)) {
+        handleValidation?.(inputVal);
```

Comment:
should we call onSegmentChange here?

Reply:
Unless there's a use case you're seeing where there's a bug, the segment change handler should already be called within `DateInputBox`